src/db/write.rs

```diff
@@ -0,0 +1,186 @@
+use std::str::FromStr;
+
+use crate::{
+    events::{Event, EventInner, ProgressionEventInner},
+    output_types::ClientEvent,
+    traits::ErrorInner,
+    Error,
+};
+
+use futures::Future;
+use log::{info, warn};
+use sqlx::{
+    sqlite::{SqliteConnectOptions, SqlitePool},
+    Pool,
+};
+use tokio::sync::broadcast::{error::RecvError, Receiver};
+
+use super::types::ClientEventRow;
+
+// TODO clean up all unwraps
+
+pub fn write_event_to_db_task(
+    mut event_receiver: Receiver<Event>,
+    sqlite_pool: SqlitePool,
+) -> impl Future<Output = ()> {
```

Comment:
async function written explicitly, this function can be written with `async` keyword